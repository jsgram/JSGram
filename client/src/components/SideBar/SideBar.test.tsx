import React from 'react';
import ConnectedSideBar from './index';
import {mount, shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import sinon from 'sinon';

describe('SideBar smart component', () => {
    let renderer: any;
    const mockStore = configureStore();
    const props = {
        urlUsername: 'somevalue',
    };
    const initialState = {
        user: 'somevalue',
        loading: true,
        profileEdit: {
            newUsername: 'somevalue',
            newFullName: 'somevalue',
        },
    };

    beforeEach(() => {
        const store = mockStore(initialState);
        renderer = shallow(<Provider store={store}><ConnectedSideBar {...props} /></Provider>);
    });

    test('render-success', () => {
        expect(renderer.find(ConnectedSideBar).length).toEqual(1);
    });
});
