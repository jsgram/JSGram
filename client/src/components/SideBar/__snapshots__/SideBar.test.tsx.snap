// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar smart component render - success 1`] = `ShallowWrapper {}`;
